frontend/src/components/dashboard/create/common/BodyTemplate.tsx

```diff
@@ -11,40 +11,49 @@ import {
   StepSection,
 } from 'components/common'
 import SaveDraftModal from 'components/dashboard/create/save-draft-modal'
-import { exceedsCharacterThreshold, saveTemplate } from 'services/sms.service'
+import { saveTemplate } from 'services/sms.service'
 
 import type { Dispatch, SetStateAction } from 'react'
-import type { SMSProgress } from 'classes'
 
 import styles from '../Create.module.scss'
 
-const SMSTemplate = ({
+function BodyTemplate({
   setActiveStep,
+  warnCharacterCount,
+  errorCharacterCount,
 }: {
-  setActiveStep: Dispatch<SetStateAction<SMSProgress>>
-}) => {
+  setActiveStep: Dispatch<SetStateAction<number>> // todo: look into tightening the type defn to SMSProgress | TelegramProgress
+  warnCharacterCount: number
+  errorCharacterCount: number
+}) {
   const { campaign, updateCampaign } = useContext(CampaignContext)
   const { setFinishLaterContent } = useContext(FinishLaterModalContext)
   const [body, setBody] = useState(replaceNewLines(campaign.body))
-  const [errorMsg, setErrorMsg] = useState(null)
+  const [errorMsg, setErrorMsg] = useState<React.ReactNode>(null)
   const { id: campaignId } = useParams<{ id: string }>()
 
   useEffect(() => {
-    if (exceedsCharacterThreshold(body)) {
-      setErrorMsg(
-        (
-          <span>
-            Your template has more than 1000 characters. Messages which are
-            longer than <b>1600</b> characters (including keywords) can&apos;t
-            be sent. Consider making your message short and sweet to make it
-            easier to read on a mobile device.
-          </span>
-        ) as any
```

Comment:
We don't need the `as any` anymore after adding a type definition for the `useState` call in line 32